schematic/models/metadata.py

```diff
@@ -315,121 +317,113 @@ def populateModelManifest(
+            val_errors, _ = self.validateModelManifest(
                 manifestPath=manifest_path,
                 rootNode=validate_component,
                 restrict_rules=restrict_rules,
                 project_scope=project_scope,
                 access_token=access_token,
             )
 
-            # if there are no errors in validation process
-            if val_errors == []:
-                # upload manifest file from `manifest_path` path to entity with Syn ID `dataset_id`
-                if os.path.exists(censored_manifest_path):
-                    censored_manifest_id = syn_store.associateMetadataWithFiles(
-                        dmge=self.dmge,
-                        metadataManifestPath=censored_manifest_path,
-                        datasetId=dataset_id,
-                        manifest_record_type=manifest_record_type,
-                        hideBlanks=hide_blanks,
-                        table_manipulation=table_manipulation,
-                        table_column_names=table_column_names,
-                        annotation_keys=annotation_keys,
-                        file_annotations_upload=file_annotations_upload,
-                    )
-                    restrict_maniest = True
-
-                manifest_id = syn_store.associateMetadataWithFiles(
-                    dmge=self.dmge,
-                    metadataManifestPath=manifest_path,
-                    datasetId=dataset_id,
-                    manifest_record_type=manifest_record_type,
-                    hideBlanks=hide_blanks,
-                    restrict_manifest=restrict_maniest,
-                    table_manipulation=table_manipulation,
-                    table_column_names=table_column_names,
-                    annotation_keys=annotation_keys,
-                    file_annotations_upload=file_annotations_upload,
-                )
-
-                logger.info(f"No validation errors occured during validation.")
-                return manifest_id
-
-            else:
+            if val_errors != []:
                 raise ValidationError(
                     "Manifest could not be validated under provided data model. "
                     f"Validation failed with the following errors: {val_errors}"
                 )
 
-        # no need to perform validation, just submit/associate the metadata manifest file
-        if os.path.exists(censored_manifest_path):
-            censored_manifest_id = syn_store.associateMetadataWithFiles(
+            # if there are no errors in validation process
+            # upload manifest file from `manifest_path` path to entity with Syn ID `dataset_id`
+            if os.path.exists(censored_manifest_path):
+                restrict_maniest = True
+
+            manifest_id = syn_store.associateMetadataWithFiles(
                 dmge=self.dmge,
-                metadataManifestPath=censored_manifest_path,
+                metadataManifestPath=manifest_path,
```

Comment:
why not use the `censored_manifest_path`?

Reply:
So this might be hard to tell because github messed up the diff, but this isn't a change. line 415 of the new code corresponds to line 437 of the old code.

Reply:
The diff is messed up here. Line 411 in the new code corresponds to line 402 in the old code.

Reply:
Okay, i thik whats missing is, which were previously lines 385-395
```
censored_manifest_id = syn_store.associateMetadataWithFiles(
                        dmge=self.dmge,
                        metadataManifestPath=censored_manifest_path,
                        datasetId=dataset_id,
                        manifest_record_type=manifest_record_type,
                        hideBlanks=hide_blanks,
                        table_manipulation=table_manipulation,
                        table_column_names=table_column_names,
                        annotation_keys=annotation_keys,
                        file_annotations_upload=file_annotations_upload,
                    )
  ```
  that goes within this if:
  ```
  if os.path.exists(censored_manifest_path):
  ```

Reply:
You're correct. It removed it thininign it wasn't needed since `censored_manifest_id` but it's used for its side effect so I put it back in.